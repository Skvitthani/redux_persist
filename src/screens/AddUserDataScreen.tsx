import {useDispatch} from 'react-redux';
import {StyleSheet, View} from 'react-native';
import {userData} from '../redux/actios/Action';
import React, {useEffect, useState} from 'react';
import ButtonComp from '../components/ButtonComp';
import InputTextComp from '../components/InputTextComp';
import {DealerNavigNavigationType} from '../navigation';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {
  TestIds,
  BannerAd,
  RewardedAd,
  AdEventType,
  BannerAdSize,
  InterstitialAd,
  RewardedAdEventType,
} from 'react-native-google-mobile-ads';

type AddUserDataScreen = NativeStackScreenProps<
  DealerNavigNavigationType,
  'AddUserDataScreen'
> & {
  navigation: () => void;
};

const adUnitId = __DEV__
  ? TestIds.ADAPTIVE_BANNER
  : 'ca-app-pub-xxxxxxxxxxxxx/yyyyyyyyyyyyyy';

const InterstitialAdID = __DEV__
  ? TestIds.INTERSTITIAL
  : 'ca-app-pub-xxxxxxxxxxxxx/yyyyyyyyyyyyyy';

const interstitial = InterstitialAd.createForAdRequest(InterstitialAdID, {
  keywords: ['fashion', 'clothing'],
});

const RewardedAdID = __DEV__
  ? TestIds.REWARDED
  : 'ca-app-pub-xxxxxxxxxxxxx/yyyyyyyyyyyyyy';

const rewarded = RewardedAd.createForAdRequest(RewardedAdID, {
  keywords: ['fashion', 'clothing'],
});

const AddUserDataScreen: React.FC<AddUserDataScreen> = ({navigation}) => {
  const dispatch = useDispatch();
  const [age, setAge] = useState<string | number>('');
  const [city, setCity] = useState<string | number>('');
  const [name, setName] = useState<string | number>('');
  const [interstitialLoaded, setInterstitialLoaded] = useState(false);
  const [rewardedLoaded, setRewardedLoaded] = useState(false);

  useEffect(() => {
    const unsubscribe = interstitial.addAdEventListener(
      AdEventType.LOADED,
      () => {
        setInterstitialLoaded(true);
      },
    );
    const unsubscribeLoaded = rewarded.addAdEventListener(
      RewardedAdEventType.LOADED,
      () => {
        setRewardedLoaded(true);
      },
    );
    const unsubscribeEarned = rewarded.addAdEventListener(
      RewardedAdEventType.EARNED_REWARD,
      reward => {},
    );
    rewarded.load();
    interstitial.load();
    return () => {
      unsubscribeLoaded();
      unsubscribeEarned();
      unsubscribe();
    };
  }, []);

  if (!rewardedLoaded) {
    return null;
  }

  if (!interstitialLoaded) {
    return null;
  }

  const onAddPress = () => {
    let userDetails = {
      age: age,
      name: name,
      city: city,
    };
    dispatch(userData(userDetails));
    setAge('');
    setName('');
    setCity('');
  };

  return (
    <View style={styles.container}>
      <BannerAd
        unitId={adUnitId as string}
        size={BannerAdSize.ANCHORED_ADAPTIVE_BANNER}
      />
      <InputTextComp
        value={name}
        placeholder="Enter Name"
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setName(txt)}
      />
      <InputTextComp
        value={age}
        placeholder="Enter Age"
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setAge(txt)}
      />
      <InputTextComp
        value={city}
        placeholder="Enter City"
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setCity(txt)}
      />
      <View style={styles.buttonView}>
        <ButtonComp onPress={onAddPress} title="Add" />
      </View>
      <View style={styles.buttonView}>
        <ButtonComp
          onPress={() => {
            navigation.navigate('GetUserData');
          }}
          title="Go to next screen"
        />
      </View>
      <View style={styles.buttonView}>
        <ButtonComp
          onPress={() => {
            interstitial.show();
          }}
          title="Show Interstitial"
        />
      </View>
      <View style={styles.buttonView}>
        <ButtonComp
          onPress={() => {
            rewarded.show();
          }}
          title="Show Rewarded Ad"
        />
      </View>
    </View>
  );
};

export default AddUserDataScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#3A4D39',
  },
  inputStyle: {
    padding: 10,
    fontSize: 15,
    color: 'black',
    marginTop: 10,
    marginHorizontal: 10,
  },
  buttonView: {
    marginTop: 20,
  },
});
