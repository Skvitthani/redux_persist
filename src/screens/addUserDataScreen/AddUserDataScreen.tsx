import React, {useState} from 'react';
import {useDispatch} from 'react-redux';
import {StyleSheet, View} from 'react-native';
import {userData} from '../../redux/actios/Action';
import ButtonComp from '../../components/ButtonComp';
import InputTextComp from '../../components/InputTextComp';
import {DealerNavigNavigationType} from '../../navigation';
import {NativeStackScreenProps} from '@react-navigation/native-stack';

type AddUserDataScreen = NativeStackScreenProps<
  DealerNavigNavigationType,
  'AddUserDataScreen'
> & {
  navigation: () => void;
};

const AddUserDataScreen: React.FC<AddUserDataScreen> = ({navigation}) => {
  const dispatch = useDispatch();
  const [age, setAge] = useState<string | number>('');
  const [city, setCity] = useState<string | number>('');
  const [name, setName] = useState<string | number>('');

  const onAddPress = () => {
    let userDetails = {
      age: age,
      name: name,
      city: city,
    };
    dispatch(userData(userDetails));
    setAge('');
    setName('');
    setCity('');
  };

  return (
    <View style={styles.container}>
      <InputTextComp
        inputTestID="name_text_input"
        value={name}
        placeholder="Enter Name"
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setName(txt)}
      />
      <InputTextComp
        inputTestID="age_text_input"
        value={age}
        placeholder="Enter Age"
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setAge(txt)}
      />
      <InputTextComp
        inputTestID="city_text_input"
        value={city}
        placeholder="Enter City"
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setCity(txt)}
      />
      <View style={styles.buttonView}>
        <ButtonComp
          buttonCompTestCase="add_user_data"
          onPress={onAddPress}
          title="Add"
        />
      </View>
      <View style={styles.buttonView}>
        <ButtonComp
          buttonCompTestCase="get_user_data"
          onPress={() => {
            navigation.navigate('GetUserData');
          }}
          title="Go to next screen"
        />
      </View>
    </View>
  );
};

export default AddUserDataScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#3A4D39',
  },
  inputStyle: {
    padding: 10,
    fontSize: 15,
    color: 'black',
    marginTop: 10,
    marginHorizontal: 10,
  },
  buttonView: {
    marginTop: 20,
  },
});
