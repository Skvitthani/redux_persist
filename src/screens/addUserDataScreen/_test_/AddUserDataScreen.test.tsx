import 'react-native';
import React from 'react';
import AddUserDataScreen from '../AddUserDataScreen';
import {fireEvent, render} from '@testing-library/react-native';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue({
    repositories: {
      loading: false,
      repos: [],
    },
  }),
  useDispatch: () => mockDispatch,
}));

const component = (
  <AddUserDataScreen
    navigation={
      {
        navigate: jest.fn(),
        goBack: jest.fn(),
      } as any
    }
    route={undefined as any}
  />
);

describe('AddUserDataScreen', () => {
  it('renders correctly', () => {
    const container = render(component);
    expect(container.toJSON()).toMatchSnapshot();
  });

  it('add user data', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('add_user_data');
    fireEvent.press(container);
  });

  it('get user data', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('get_user_data');
    fireEvent.press(container);
  });

  it('name text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('name_text_input');
    fireEvent.changeText(container);
  });

  it('age text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('age_text_input');
    fireEvent.changeText(container);
  });

  it('city text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('city_text_input');
    fireEvent.changeText(container);
  });
});
