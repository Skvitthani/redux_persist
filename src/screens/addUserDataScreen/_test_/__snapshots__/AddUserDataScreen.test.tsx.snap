// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserDataScreen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#3A4D39",
      "flex": 1,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Name"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "color": "black",
          "fontSize": 15,
          "marginHorizontal": 10,
          "marginTop": 10,
          "padding": 10,
        },
      ]
    }
    testID="name_text_input"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Age"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "color": "black",
          "fontSize": 15,
          "marginHorizontal": 10,
          "marginTop": 10,
          "padding": 10,
        },
      ]
    }
    testID="age_text_input"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter City"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "color": "black",
          "fontSize": 15,
          "marginHorizontal": 10,
          "marginTop": 10,
          "padding": 10,
        },
      ]
    }
    testID="city_text_input"
    value=""
  />
  <View
    style={
      {
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706233",
          "borderRadius": 20,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="add_user_data"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Add
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706233",
          "borderRadius": 20,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="get_user_data"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Go to next screen
      </Text>
    </View>
  </View>
</View>
`;
